File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        currentSum = 0
        
        for x in nums:
            # negative numbers won't contribute to a maximum sum
            if currentSum < 0:
                currentSum = 0
            currentSum += x
            maxSub = max(maxSub, currentSum)
        
        return maxSub

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSub = nums[0];
        int currentSum = 0;

        for(int x: nums){
            if(currentSum < 0){
                currentSum = 0;
            }
            currentSum += x;
            maxSub = max(maxSub, currentSum);
        }
        
        return maxSub;
    }
};